core/src/main/java/com/scalar/db/util/groupcommit/DelayedSlotMoveWorker.java

```diff
@@ -20,11 +23,17 @@ class DelayedSlotMoveWorker<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY, V>
         queueCheckIntervalInMillis,
         // Enqueued items of this worker can be out of order since `delayedSlotMoveTimeoutAt` would
         // be updated.
-        RetryMode.MOVE_TO_TAIL);
+        RetryMode.RE_ENQUEUE);
     this.groupManager = groupManager;
     this.groupCleanupWorker = groupCleanupWorker;
   }
 
+  @Override
+  BlockingQueue<NormalGroup<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY, V>> createQueue() {
+    return new PriorityBlockingQueue<>(
```

Comment:
Does this mean an item with the smallest timeout value is taken first?
It might be helpful to add a comment on that.

Reply:
Thanks for the feedback! Addressed it in https://github.com/scalar-labs/scalardb/pull/1641/commits/7e66c31d4ffa168c47d2952d7b3344b579f38148